Integrate new Broker metric exporter parameters

Neverlord opened the pull request:
Integrates https://github.com/zeek/broker/pull/183 and adds Zeek-side adjustments to allow configuring the metric exporter at runtime (via options). Filing as draft for two reasons:
1. I'm looking for feedback on the script land integration. Is calling the BIFs via change handlers the best way to go about it? Also, I'm not sure how I could write tests for this. So it'd be great if someone (@0xxon? 🙂) would play with it a bit.
2. There's one piece missing that I'm unsure how to implement it: during startup, Broker initializes its state via defaults, config files and environment variables. Is there a way to initialize the options from C++ after Broker is done initializing? Ideally without triggering the change handlers, but wouldn't be much of an issue if they fire anyways.

jsiwek commented on scripts/base/frameworks/broker/main.zeek:
```diff
@@ -385,9 +404,41 @@ event Broker::log_flush() &priority=10
 event zeek_init()
 	{
 	schedule Broker::log_batch_interval { Broker::log_flush() };
+	Option::set_change_handler("Broker::metrics_exporter_interval",
```
Based on description of change handlers from https://docs.zeek.org/en/master/frameworks/configuration.html#when-change-handlers-trigger, think an explicit call to each individual `update_metrics_` function is warranted at this point to ensure those actually consistently get set at startup-time to what a user expects.

Neverlord replied:
Done.

jsiwek reviewed (commented):
> Is calling the BIFs via change handlers the best way to go about it?

Yes, I think so, though I had one comment about adding explicit calls to the `update_metrics_` functions from within `zeek_init()` to ensure the settings are consistently applied at startup-time based on the initial `option` values rather than only when altered at run-time.

> There's one piece missing that I'm unsure how to implement it: during startup, Broker initializes its state via defaults, config files and environment variables. Is there a way to initialize the options from C++ after Broker is done initializing?

Assuming there's appropriate getter-methods to replace the `...` below with the actual value Broker is using, I think you can sync the initial value of each Zeek `option` like this toward the end of `zeek::Broker::Manager::InitPostScript()`:

```c++
auto id = zeek::id::find("Broker::metrics_exporter_target");
id->SetVal(...);
```

May also want to update the comments for the `option` variables to explain their initial values may be overridden by config file or specific environment variables.

Neverlord commented on scripts/base/frameworks/broker/main.zeek:
```diff
@@ -122,6 +122,25 @@ export {
+	option metrics_exporter_interval = 1 sec;
+
+	## Target topic for the metrics. Setting a non-empty string starts the
+	#periodic publishing of local metrics.
```
Sorry about that, I'm so used to VIM adding proper documentation markers when hitting enter that I didn't notice this.

Neverlord commented:
Thanks a lot for the feedback and pointers! I've added the extra synchronization steps to `InitPostScript` and also fixed some naming inconsistencies between the option names and the environment variable names.

Neverlord commented:
CI is only red because LGTM complains about merge conflicts. @jsiwek should I rebase the branch or can I ignore it since you need to touch `auxil/broker` anyway after merging the Broker PR?

jsiwek commented on src/broker/comm.bif:
```diff
@@ -145,3 +145,45 @@ function Broker::__node_id%(%): string
+	%{
+	zeek::Broker::Manager::ScriptScopeGuard ssg;
+	if ( broker_mgr )
+	{
+		std::vector<std::string> slist;
+		auto* vval = filter->AsVectorVal();
+		for ( unsigned index = 0; index < vval->Size(); ++index )
+			slist.emplace_back(vval->StringValAt(index)->ToStdString());
+		broker_mgr->SetMetricsExportPrefixes(std::move(slist));
+	}
```
Nitpick: missing a tab-indent on the braces.

Neverlord replied:
Yep, indented properly now.

jsiwek reviewed (commented):
> should I rebase the branch or can I ignore it since you need to touch `auxil/broker` anyway after merging the Broker PR?

Thanks for checking, a rebase (or just merge) would be fine since I may end up just approving and then giving time to let someone else be final judge/merger after they experiment with using things.

Neverlord commented on src/broker/Manager.cc:
```diff
@@ -41,6 +41,86 @@ static inline Val* get_option(const char* option)
+		auto ptr = make_intrusive<VectorVal>(zeek::id::string_vec);
+		for ( const auto& str : value )
+			ptr->Append(make_intrusive<StringVal>(str));
+		}
```
Fixed.

Neverlord commented:
> > should I rebase the branch or can I ignore it since you need to touch `auxil/broker` anyway after merging the Broker PR?
> 
> Thanks for checking, a rebase (or just merge) would be fine since I may end up just approving and then giving time to let someone else be final judge/merger after they experiment with using things.

Ok. Thanks for the review again! I've merged the master to the branch and cleaned up the remaining issues.
